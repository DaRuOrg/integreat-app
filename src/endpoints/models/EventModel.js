export default class EventModel {
  constructor ({ id, title = '', content = '', thumbnail = '', address = '', town = '', date = {} }) {
    this._id = id
    this._title = title
    this._content = content
    this._thumbnail = thumbnail
    this._address = address
    this._town = town
    this._date = date
  }

  get id () {
    return this._id
  }

  get title () {
    return this._title
  }

  get thumbnail () {
    return this._thumbnail
  }

  get content () {
    return this._content
  }

  get address () {
    return this._address + ', ' + this._town
  }

  getDate (locale) {
    return this._date.toLocaleString(locale)
  }
}
