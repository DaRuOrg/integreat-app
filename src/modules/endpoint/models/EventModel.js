// @flow

import moment from 'moment'

type eventModel = {
  id: boolean,
  title: string,
  content: string,
  thumbnail: string,
  availableLanguages: any,
  content: string,
  town: string,
  address: string,
  startDate: moment,
  endDate: moment,
  allDay: boolean,
  excerpt: string
}

class EventModel {
  _id: boolean
  _title: string
  _content: string
  _thumbnail: string
  _availableLanguages: any
  _content: string
  _town: string
  _address: string
  _startDate: moment
  _endDate: moment
  _allDay: boolean
  _excerpt: string

  constructor (eventModel: eventModel) {
    this._id = eventModel.id
    this._title = eventModel.title
    this._content = eventModel.content
    this._thumbnail = eventModel.thumbnail
    this._address = eventModel.address
    this._town = eventModel.town
    this._startDate = eventModel.startDate
    this._endDate = eventModel.endDate
    this._allDay = eventModel.allDay
    this._excerpt = eventModel.excerpt
    this._availableLanguages = eventModel.availableLanguages
  }

  get id (): boolean {
    return this._id
  }

  get title (): string {
    return this._title
  }

  get thumbnail (): string {
    return this._thumbnail
  }

  get content (): string {
    return this._content
  }

  get address (): ?string {
    if (this._address && this._town) {
      return `${this._address}, ${this._town}`
    } else if (this._address) {
      return this._address
    } else if (this._town) {
      return this._town
    }
  }

  get startDate (): moment {
    return this._startDate
  }

  get endDate (): moment {
    return this._endDate
  }

  get allDay (): boolean {
    return this._allDay
  }

  get excerpt (): string {
    return this._excerpt
  }

  get availableLanguages (): any {
    return this._availableLanguages
  }
}

export default EventModel
