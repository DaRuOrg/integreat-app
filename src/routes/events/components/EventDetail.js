import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'

import EventModel from 'modules/endpoint/models/EventModel'
import RemoteContent from 'modules/common/components/RemoteContent'

import style from './EventDetail.css'
import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import Caption from 'modules/common/components/Caption'

/**
 * Display a single event with all necessary information
 */
class EventDetail extends React.Component {
  static propTypes = {
    event: PropTypes.instanceOf(EventModel).isRequired,
    language: PropTypes.string.isRequired
  }

  getEventPlaceholder () {
    const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
    return placeholders[this.props.event.id % 3]
  }

  render () {
    const {t} = this.props
    return (
      <div>
        <img className={style.thumbnail} src={this.props.event.thumbnail || this.getEventPlaceholder()} />
        <Caption title={this.props.event.title} />
        <div>
          <span className={style.identifier}>{t('date')}: </span>
          <span className={style.date}>
            {this.props.event.dateModel.toLocaleString(this.props.language)}
          </span>
        </div>
        <div>
          <span className={style.identifier}>{t('location')}: </span>
          <span className={style.date}>{this.props.event.address}</span>
        </div>
        <RemoteContent dangerouslySetInnerHTML={{__html: this.props.event.content}} />
      </div>
    )
  }
}

export default translate('events')(EventDetail)
