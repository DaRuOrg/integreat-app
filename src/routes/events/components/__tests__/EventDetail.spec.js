import React from 'react'
import { shallow } from 'enzyme'

import EventModel from 'modules/endpoint/models/EventModel'
import EventDetail from '../EventDetail'

describe('EventDetail', () => {
  const event = new EventModel({
    id: 1234,
    title: 'first Event',
    availableLanguages: {de: '1235', ar: '1236'},
    startDate: new Date('2017-11-18 09:30:00'),
    endDate: new Date('2017-11-18 19:30:00'),
    allDay: true
  })

  const language = 'en'

  test('should render', () => {
    expect(shallow(<EventDetail event={event} language={language} />)).toMatchSnapshot()
  })
})
