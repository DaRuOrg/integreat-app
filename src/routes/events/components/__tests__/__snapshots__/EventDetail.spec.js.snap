// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetail should render 1`] = `
<EventDetail
  event={
    EventModel {
      "_address": "",
      "_availableLanguages": Object {
        "ar": "1236",
        "de": "1235",
      },
      "_content": "",
      "_date": Object {},
      "_excerpt": "",
      "_id": 1234,
      "_thumbnail": "",
      "_title": "first Event",
      "_town": "",
    }
  }
  language="en"
  t={[Function]}
/>
`;
